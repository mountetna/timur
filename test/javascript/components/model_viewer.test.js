import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { mockStore } from '../helpers';
import ModelViewer from '../../../lib/client/jsx/components/model_viewer';

const MONSTER = require('../fixtures/template_monster.json');

describe('ModelViewer', () => {
  let store;
  let template;

  beforeEach(() => {
    store = mockStore({
      location: {
        path: '/labors/browse/monster/Nemean Lion'
      },
      directory: {
        uploads: {}
      },
      magma: {
        models: {
          monster: {
            template: MONSTER,
            documents: {
              'Nemean Lion': { name: 'Nemean Lion', species: 'lion' },
              'Lernean Hydra': { name: 'Lernean Hydra', species: 'hydra' }
            }
          }
        }
      }
    });
  });

  it('renders', () => {
    global.CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModelViewer
            model_name={'monster'}
            record_names={ ['Nemean Lion', 'Lernean Hydra'] }
        />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
