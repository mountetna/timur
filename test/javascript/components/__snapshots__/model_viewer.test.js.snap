// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelViewer renders 1`] = `
<div
  className="table"
>
  <div
    className="ReactTable"
    style={Object {}}
  >
    <div
      className="rt-table"
      role="grid"
    >
      <div
        className="rt-thead -header"
        style={
          Object {
            "minWidth": "700px",
          }
        }
      >
        <div
          className="rt-tr"
          role="row"
        >
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              aspect
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              labor
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              name
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              species
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              stats
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              certificates
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
          <div
            className="rt-th rt-resizable-header"
            onClick={[Function]}
            role="columnheader"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
            tabIndex="-1"
          >
            <div
              className="rt-resizable-header-content"
            >
              victim
            </div>
            <div
              className="rt-resizer"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="rt-tbody"
        style={
          Object {
            "minWidth": "700px",
          }
        }
      >
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -odd"
            role="row"
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              />
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                Lernean Hydra
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                hydra
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute file"
              >
                <span
                  className="file-missing"
                >
                   No file 
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="rt-tr-group"
          role="rowgroup"
        >
          <div
            className="rt-tr -even"
            role="row"
          >
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              />
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                Nemean Lion
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                lion
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute file"
              >
                <span
                  className="file-missing"
                >
                   No file 
                </span>
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
            <div
              className="rt-td"
              onClick={[Function]}
              role="gridcell"
              style={
                Object {
                  "flex": "100 0 auto",
                  "maxWidth": null,
                  "width": "100px",
                }
              }
            >
              <div
                className="attribute"
              >
                <div
                  className="collection"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="-loading"
    >
      <div
        className="-loading-inner"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
